src/storage/mvcc/reader/backward_scanner.rs

```diff
@@ -0,0 +1,713 @@
+                last_checked_commit_ts = Key::decode_ts_from(current_key)?;
+
+                if !Key::is_user_key_eq(current_key, user_key.encoded().as_slice())
+                    || last_checked_commit_ts > safe_ts
+                {
+                    // Meet another key or meet an unwanted version. We use `last_version` as the return.
+                    is_done = true;
+                }
+            }
+            if is_done {
+                return Ok(self.handle_last_version(last_version, user_key)?);
+            }
+
+            let write = Write::parse(self.write_cursor.value(&mut self.statistics.write))?;
+            self.statistics.write.processed += 1;
+
+            match write.write_type {
+                WriteType::Put | WriteType::Delete => last_version = Some(write),
+                WriteType::Lock | WriteType::Rollback => {}
+            }
+        }
+
+        // At this time, we must have current commit_ts <= safe_ts. If commit_ts == safe_ts,
+        // we don't need to seek any more and we can just utilize `last_version`.
+        if last_checked_commit_ts == safe_ts {
+            return Ok(self.handle_last_version(last_version, user_key)?);
+        }
+
+        // After several `prev()`, we still not get the latest version for the specified ts,
+        // use seek to locate the latest version.
+        let last_handled_key = self.write_cursor.key(&mut self.statistics.write).to_vec();
+
+        let seek_key = user_key.clone().append_ts(safe_ts);
+        assert!(seek_key.encoded().as_slice() < last_handled_key.as_slice());
+
+        // TODO: Replace by cast + seek().
+        self.write_cursor
+            .internal_seek(&seek_key, &mut self.statistics.write)?;
+        assert!(self.write_cursor.valid());
+
+        loop {
+            // After seek, or after some `next()`, we may reach `last_handled_key` again. It
+            // means we have checked all versions for this user key. We use `last_version` as
+            // return.
+            let mut is_done = false;
+            {
+                let current_key = self.write_cursor.key(&mut self.statistics.write);
+                // We should never reach another user key.
+                assert!(Key::is_user_key_eq(
+                    current_key,
+                    user_key.encoded().as_slice()
+                ));
+                if current_key >= last_handled_key.as_slice() {
+                    // We reach the last handled key,
+                    is_done = true;
+                }
+            }
+            if is_done {
+                return Ok(self.handle_last_version(last_version, user_key)?);
+            }
+
+            let write = Write::parse(self.write_cursor.value(&mut self.statistics.write))?;
+            self.statistics.write.processed += 1;
+
+            match write.write_type {
+                WriteType::Put => {
+                    return Ok(Some(self.reverse_load_data_by_write(write, user_key)?))
```

Comment:
When loading data, we can inspired by the way how we get the commit information. If we use one or two times `prev` and get the commit information, we can also use prev when loading data, if we use several `prev` and `seek` to get the commit information, we can use seek directly when loading data.

Reply:
Aha. Interesting thought. Worth to have a try.